### PR TITLE
Add support to the most recent PHP versions

### PR DESCRIPTION
**Pull request by henriquemoody**
Hi there!

I've noticed that the library does not run in PHP 8.4, so I add the latest PHP versions and made some fixes, however, it seems like there's quite a lot to change.

* Are you planning to give support for the most recent PHP versions any time soon?
* Are you planning on dropping support to old PHP versions?

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -14,14 +14,13 @@
...
-        "phpunit/phpunit": ">=7.5.20",
         "php-coveralls/php-coveralls": "^2.1",
+        "phpmd/phpmd": "@stable",
+        "phpunit/phpunit": "^9.6",
+        "sokil/php-isocodes-db-i18n": "^4.0.0",
         "squizlabs/php_codesniffer": "^3.4",
-        "phpmd/phpmd" : "@stable",
         "symfony/translation": "^4.4.17|^5.2",
-        "vimeo/psalm": "^4.3",
-        "infection/infection": ">=0.11.5"
```
Comment (henriquemoody) [inline composer.json]: there are several error messages coming from Infection

File: tests/docker/docker-php-entrypoint [hunk 2]
```diff
@@ -14,4 +14,4 @@ echo "uk_UA.UTF-8 UTF-8" > /etc/locale.gen
 locale-gen
 
 # run tests
-/php-isocodes/vendor/bin/phpunit -c /php-isocodes/tests/phpunit.xml /php-isocodes/tests
\ No newline at end of file
+/php-isocodes/vendor/bin/phpunit /php-isocodes/tests
```
Comment (henriquemoody) [inline tests/docker/docker-php-entrypoint]: The `./tests/docker/run-test.sh` didn't work as stated in the documentation.

File: .github/workflows/continuous-integration.yml [hunk 3]
```diff
@@ -13,12 +13,10 @@ jobs:
...
         php-version:
-          - "7.1"
-          - "7.2"
-          - "7.3"
-          - "7.4"
-          - "8.0"
           - "8.1"
+          - "8.2"
+          - "8.3"
+          - "8.4"
```
Comment (sokil) [inline .github/workflows/continuous-integration.yml:24]: Good new versions but library is used by 10% of php 7.4 and 8.0.
I may drop support only with major version increment and currently i have no reason to do this. 
If you return old versions i may merge this )

Comment (henriquemoody) [reply .github/workflows/continuous-integration.yml:24]: The problem is that by keeping those old PHP versions we can't update other packages, it's a chicken and egg thing. So, you're gonna have to make a choice 🙃

Comment (sokil) [reply .github/workflows/continuous-integration.yml:24]: These are dev dependencies. is i understand some of them do not accept php 8.4? 
I will check this soon.
thanks

Comment (henriquemoody) [reply .github/workflows/continuous-integration.yml:24]: Oh, never mind, it's actually working with PHP 7.4.

The problem is that I had to remove psalm, they're still working on it https://github.com/vimeo/psalm/issues/11107

Comment (henriquemoody) [reply .github/workflows/continuous-integration.yml:23]: I couldn't make it work from version <=7.2

**Comment by henriquemoody** [comment]
I reverted the changes I made, and just added the support for the new versions.

**Comment by henriquemoody** [comment]
Hi @sokil! When would you release the patch version?

I'm waiting for it to add support to PHP 8.4 to [Respect\Validation](https://github.com/Respect/Validation)

**Comment by sokil** [comment]
Check 4.2.1

Thank you

**Comment by henriquemoody** [comment]
I can't find version 4.2.1, yet, when are you planning on creating it?

**Comment by sokil** [comment]
Sorry, version is on the way to packagist. Please, check.

**Comment by henriquemoody** [comment]
Thank you!

**Pull request merged**